projects/plugins/jetpack/_inc/client/components/navigation-settings/index.jsx

```diff
@@ -37,9 +37,14 @@ export class NavigationSettings extends React.Component {
 	};
 
 	UNSAFE_componentWillMount() {
-		// We need to handle the search term not only on route update but also on page load in case of some external redirects
 		this.onRouteChange( this.props.location );
-		this.props.history.listen( this.onRouteChange );
+	}
+
+	componentDidUpdate( oldprops ) {
+		// We need to handle the search term not only on route update but also on page load in case of some external redirects
+		if ( oldprops.location !== this.props.location ) {
+			this.onRouteChange( this.props.location );
+		}
```

Comment:
There's no direct upgrade path for this, but hopefully this roughly the same thing as before.